lib/hardware/led_light.hpp

```diff
@@ -0,0 +1,17 @@
+#ifndef PET_PASSAGE_HARDWARE_LED_LIGHT_H_
+#define PET_PASSAGE_HARDWARE_LED_LIGHT_H_
+
+#include "Arduino.h"
+#include "led_base.hpp"
+
+template <uint8_t Pin>
+class LedLight : public LedBase {
+ public:
+  void virtual init();
```

Comment:
Instead of declaring these virtual you should use the (override)[https://en.cppreference.com/w/cpp/language/override] specifier